docs/_pages/general/rooting-instructions.md

```diff
@@ -7,6 +7,114 @@ order: 9
+
+Now you have to open a serial connection from your laptop to the device, this can be done with putty, miniterm, minicom or through a tool like screen with the following command: `screen /dev/ttyUSB0 115200,ixoff`. The baud rate is 115200 and flow control (XIN, XOUT) needs to be off.
+
+Once your connection is ready, turn on the vacuum by pressing and holding the middle button (POWER) for at least 3 seconds. 
+
+You should see some logs and one of the last ones will say root password changed. If your logs are going crazy, one of your wires are probably loose or doesn't have a good connection.
+
+To use the Wifi Reset method, open up the other side of the robot and press the reset button shortly (<1 second) with a pen or paperclip. Your UART connection should pop up with the login prompt like `"p2029_release login”`
+
+When connected, you can log in as `root` and then it will ask for a passwrod.
+To calculate the password use the full serial number of your robot, which can be found on the sticker below the dustbin.
+**Not the one on the bottom of the robot nor the one on the packaging. You'll have to take out the dustbin and look below it into the now empty space.**
+
+To get the password, use the following [Calculator](https://gchq.github.io/CyberChef/#recipe=Find_/_Replace(%7B'option':'Regex','string':'(%5C%5Cn%7C%5C%5Cr)'%7D,'',true,false,true,false)MD5()Find_/_Replace(%7B'option':'Regex','string':'$'%7D,'%20%20-%5C%5Cn',false,false,false,false)To_Base64('A-Za-z0-9%2B/%3D')&input=UDIwMDkwMDAwRVUwMDAwMFpN) or enter the full SN (all uppercase) into this command on Linux
+`echo -n "P20290000US00000ZM" | md5sum | base64` or the following commands on Mac 
+````
+echo -n "P20290000US00000ZM" | md5
+echo -n -e "MD5HASHONLY"  -\n" | base64
+````
+
+Once logged in, build a patched firmware image for manual installation via the [Dustbuilder](http://dustbuilder.dontvacuum.me/_dreame_p2029.html). You’ll need to put in your email, serial number and SSH key if you have one. Make sure you settings match these
+
+✅Patch DNS (requirement for valetudo deployment, disables real cloud!!)
+
+✅Preinstall Nano texteditor, curl, wget, htop, hexdump 
+
+✅Build for manual installation (requires SSH to install)
+
+Then accept at the bottom and `Create Job`. This will send your build to your email once it’s built. Download the `tar.gz` file to your laptop.
+
+To get this file over to the robot, you'll need to spin up a temporary webserver (e.g. by using `python3 -m http.server`) in the directory where you downloaded your firmware image to, connect the laptop to the robots WiFi access point and download the firmware image to the robot via e.g. `wget http://<your-laptop-ip>/dreame.vacuum.p2029_fw.tar.gz`. If you're running Home Assistant, you might find it easier to put the file in your `/config/www/` folder via Samba Share and then grab the file over http from there. For Home Assistant method, you need to replace `<your-laptop-ip>` with `<your-ha-ip-address>:8123/local/`
```

Comment:
This likely refers to hassio which is the appliance version of home assistant and not home assistant in general